File: benchmarks/c/fib.c
// Equivalent C impl. Build: cc -O2 fib.c -o fib_c
#include <stdio.h>
#include <stdint.h>

int64_t fib(int64_t n) {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

int main(void) {
    int64_t n = 35;
    int64_t r = fib(n);
    printf("fib(35) = %lld\n", (long long)r);
    return 0;
}

File: benchmarks/rust/fib.rs
// Equivalent Rust impl. Build: rustc -O fib.rs -o fib_rs

fn fib(n: i64) -> i64 {
    if n < 2 {
        return n;
    }
    fib(n - 1) + fib(n - 2)
}

fn main() {
    let n: i64 = 40;
    let r = fib(n);
    println!("fib(40) = {}", r);
}
